modules/internal/ChapelArray.chpl

```diff
@@ -167,6 +167,9 @@ module ChapelArray {
   use ArrayViewRankChange;
   use ArrayViewReindex;
 
+  pragma "no doc"
+  config param showArrayAsVectorWarnings = false;
```

Comment:
Is this a missed change? The remainder of this file uses `showArrayAsVectorDeprecations`. Maybe that's why the checks are failing at the moment.

Reply:
Also, is it supposed to be `false` by default?

Reply:
The original PR had the checks always-on.  The config was added so programmers could opt-in / out.  The goal is to release with this `true` but this PR may be merged with it in the `false` condition if we don't have clean testing (if we do, it should be changed back to `true`).

Reply:
One thing that is still yet to be decided is what to do with all the tests written specifically for "array-as-vec" operations. If we want to merge this PR with the config ON, then we will have to do one of the following for all those tests (where _currently_ they are unchanged):

- Update their good files
- Remove the test
- Migrate the test to `test/deprecated`

Reply:
My 2¢ for tests specifically for array-as-vec: update their good files now, and when the deprecated code is removed, make sure that tests of functionality that exists in `list` are moved over to `list`.

Reply:
I agree with @mppf. The ideal thing to do in the end is to update the tests to do what they do now, but using `list` functionality.

Reply:
I disagree that we should put off rewriting tests to use list until the array-is-vec features are removed.  If the tests exercise array-is-vec functionality that also exists on list, I think those tests should be rewritten now to put more weight on list since that's the preferred solution and the one we want to make sure is at least as good as array-is-vec was.  If existing list tests already covers the test's functionality sufficiently that the rewriting is unnecessary, I think the test should just be moved to test/deprecated (with updated .good file).  If the test simply doesn't make sense in a list world (e.g., queries the domain of an array after a number of array-as-vec operations), I'd also move it to test/deprecated.